das/checkpoint.go

```diff
@@ -48,3 +48,12 @@ func (c checkpoint) String() string {
 
 	return str
 }
+
+// totalSampled returns the total amount of sampled headers
+func (c checkpoint) totalSampled() int {
```

Comment:
What if `totalSampled < 0`?
Can we make the return value `uint64`? It will allow us to avoid converting `uint64` to `int` in this func.  We can convert it into int in`recordTotalSampled`?

Reply:
sounds good